enhancements/coreos-encrypted-by-default.md

```diff
@@ -0,0 +1,425 @@
+ 3      537MB   538MB   1049kB               BIOS-BOOT   bios_grub
+ 4      538MB   17.2GB  16.6GB  xfs          rootfs
+ ```
+
+In this idea, partition 4, will move the root XFS filesystem into a LUKS container:
+```
+Model: Unknown (unknown)
+Disk /dev/nbd0: 17.2GB
+Sector size (logical/physical): 512B/512B
+Partition Table: gpt
+Disk Flags:
+
+Number  Start   End     Size    File system  Name        Flags
+ 1      1049kB  404MB   403MB   ext4         boot
+ 2      404MB   537MB   133MB   fat16        EFI-SYSTEM  boot, esp
+ 3      537MB   538MB   1049kB               BIOS-BOOT   bios_grub
+ 4      538MB   17.2GB  16.6GB               crypt-root
+
+# blkid /dev/nbd0p4
+/dev/nbd0p4: UUID="1a7086bd-6514-457c-bc06-1215944f026d" TYPE="crypto_LUKS" PARTLABEL="root" PARTUUID="01317d04-4b83-463e-8762-5a07ec2499e6"
+```
+
+CoreOS today uses `/dev/disk/by-label` for mounting the root-filesystem. Once the LUKS container `crypt-root` is opened, `/dev/disk/by-label/rootfs` will appear.
+
+
+As an example, partition 4 could be prepared by creating a well-known plain-text password (e.g. 'coreos') using the `cipher_null`. When `cipher_null` is used, any passphrase is acceptable.
+```
+# echo "coreos" > /tmp/disk.key
+# cryptsetup luksFormat \
+    -q \
+    --label="coreos_rootfs-default" \
+    --cipher=cipher_null \
+    --key-file=/tmp/disk.key \
+    /dev/nbd0p4
+```
+
+Perform the LUKS binding (example of future command)
+```
+# clevis-luks-bind \
+       -d /dev/nbd0p4 \
+      plaintext  '{"passphrase": "doesNotMatter"}'
+```
+
+During the first boot, a Dracut module will create a new random-key and then re-encrypt the device. For example:
+```
+# head -c 256 > new_key
+# echo coreos |
+  cryptsetup-reencrypt \
+     --cipher="aes-cbc-essiv:sha256" \
+     --key-size=256 \
+     --progress-frequency=5 \
+     --key-file=new_key
+     /dev/nbd0p4
+```
+
+In the above step, a new LUKS master key created, and each block is re-encrypted. After the encryption, an appropriate Clevis configuration will be applied and the `new_key` will be removed; future encryption operations will use existing Clevis configruation.
+
+### Clevis Pins
+
+This idea hinges around Clevis, which provides an interface around different secret backend and includes a Dracut module which provides for unattended boot. Clevis uses [JSON Object Signing and Encryption, (JOSE)](https://jose.readthedocs.io/en/latest/).
+
+When Clevis binds a LUKS volume, a new JOSE object is written in the LUKS meta-data as a token. Clevis reads the JOSE object and then calls the appropriate pin. Clevis was added to Fedora 24 and RHEL 7, and it supported and recommended for unattended-encrypted boot.
+
+The first pin deals with the null case, where there is no NBE boot. This pin would be used to store an encoded plaintext password. In the event of the plaintext pin is used, the actual passphrase is obfucated (here be dragons: the axiom that "security by obscurity is not security" applies).
```

Comment:
s/NBE/NBDE/ right?

Reply:
I've read all of this and still don't understand the details for the "plaintext pin".

Reply:
Updated this stanza. See comment https://github.com/openshift/enhancements/pull/15#discussion_r326286074